### PR TITLE
[CloudKit] Update bindings to beta 6.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/cloudkit.cs [hunk 1]
```diff
@@ -489,6 +494,9 @@ interface CKErrorFields {
 
 	[iOS (8,0), Watch (3,0), TV (10,0), Mac (10,10, onlyOn64 : true)]
 	[BaseType (typeof (CKOperation))]
+	[Availability (Introduced = Platform.iOS_8_0 | Platform.Mac_10_10 | Platform.Watch_3_0 | Platform.TV_9_0,
```
Comment (spouliot) [inline src/cloudkit.cs]: The `Introduced` does not match the attributes above, just include `Deprecated`.
Also please re-write the message to follow our conventions.

File: src/cloudkit.cs [hunk 2]
```diff
@@ -758,6 +766,9 @@ interface CKLocationSortDescriptor : NSSecureCoding {
 	[iOS (8,0), Mac (10,10, onlyOn64 : true)]
 	[BaseType (typeof (CKOperation))]
 	[DisableDefaultCtor] // NSInvalidArgumentException Reason: You must call -[CKMarkNotificationsReadOperation initWithNotificationIDsToMarkRead:]
+	[Availability (Introduced = Platform.iOS_8_0 | Platform.Mac_10_10 | Platform.Watch_3_0 | Platform.TV_9_0, 
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+		Message = "Instead of iterating notifications, consider using CKDatabaseSubscription, CKFetchDatabaseChangesOperation, and CKFetchRecordZoneChangesOperation as appropriate.")]
```
Comment (spouliot) [inline src/cloudkit.cs]: same

File: src/cloudkit.cs [hunk 3]
```diff
@@ -778,6 +789,9 @@ interface CKMarkNotificationsReadOperation {
 	[iOS (8,0), Watch (3,0), TV (10,0), Mac (10,10, onlyOn64 : true)]
 	[DisableDefaultCtor] // does not work on watchOS, working stub provided to ease source compatibility
 	[BaseType (typeof (CKOperation))]
+	[Availability (Introduced = Platform.iOS_8_0 | Platform.Mac_10_10 | Platform.Watch_3_0 | Platform.TV_9_0, 
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+		Message = "No longer supported, will cease working at some point in the future.")]
```
Comment (spouliot) [inline src/cloudkit.cs]: same

File: src/cloudkit.cs [hunk 4]
```diff
@@ -1035,6 +1095,9 @@ interface CKOperation {
 		// [Export ("activityStart")]
 		// ulong ActivityStart ();
 
+		[Availability (Introduced = Platform.iOS_8_0 | Platform.Mac_10_10 | Platform.Watch_3_0 | Platform.TV_9_0, 
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+			Message = "Instead use CKOperationConfiguration.")]
```
Comment (spouliot) [inline src/cloudkit.cs]: our format is `"Use 'XXxxx' instead."`

File: src/cloudkit.cs [hunk 5]
```diff
@@ -1044,6 +1107,9 @@ interface CKOperation {
 		[Export ("usesBackgroundSession", ArgumentSemantic.UnsafeUnretained)]
 		bool UsesBackgroundSession { get; set; }
 
+		[Availability (Introduced = Platform.iOS_8_0 | Platform.Mac_10_10 | Platform.Watch_3_0 | Platform.TV_9_0, 
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+			Message = "Instead use CKOperationConfiguration.")]
```
Comment (spouliot) [inline src/cloudkit.cs]: same

File: src/cloudkit.cs [hunk 6]
```diff
@@ -1055,21 +1121,62 @@ interface CKOperation {
 		[iOS (9,3)][Mac (10,11,4)]
 		[TV (9,2)]
 		[Export ("longLived")]
+		[Availability (Introduced = Platform.iOS_9_3 | Platform.Mac_10_10 | Platform.Watch_3_0 | Platform.TV_9_2, 
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+			Message = "Instead use CKOperationConfiguration.")]
```
Comment (spouliot) [inline src/cloudkit.cs]: same

File: src/cloudkit.cs [hunk 7]
```diff
@@ -1055,21 +1121,62 @@ interface CKOperation {
...
 		bool LongLived { [Bind ("isLongLived")] get; set; }
 
 		[iOS (10,0), TV (10,0), Mac (10,12, onlyOn64 : true)]
 		[Export ("timeoutIntervalForRequest")]
+		[Availability (Introduced = Platform.iOS_10_0 | Platform.Mac_10_12 | Platform.Watch_3_0 | Platform.TV_10_0, 
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+			Message = "Instead use CKOperationConfiguration.")]
```
Comment (spouliot) [inline src/cloudkit.cs]: same

File: src/cloudkit.cs [hunk 8]
```diff
@@ -1055,21 +1121,62 @@ interface CKOperation {
...
 		double TimeoutIntervalForRequest { get; set; }
 
 		[iOS (10,0), TV (10,0), Mac (10,12, onlyOn64 : true)]
 		[Export ("timeoutIntervalForResource")]
+		[Availability (Introduced = Platform.iOS_10_0 | Platform.Mac_10_12 | Platform.Watch_3_0 | Platform.TV_10_0, 
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+			Message = "Instead use CKOperationConfiguration.")]
```
Comment (spouliot) [inline src/cloudkit.cs]: same

File: src/cloudkit.cs [hunk 9]
```diff
@@ -1055,21 +1121,62 @@ interface CKOperation {
...
 		[NullAllowed]
 		[Export ("longLivedOperationWasPersistedBlock", ArgumentSemantic.Strong)]
-		Action LongLivedOperationWasPersistedCallback { get; set; }		
+		Action LongLivedOperationWasPersistedCallback { get; set; }
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13, onlyOn64 : true), iOS (11, 0)]
+		[Export ("configuration", ArgumentSemantic.Copy)]
+		CKOperationConfiguration Configuration { get; set; }
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
```
Comment (spouliot) [inline src/cloudkit.cs]: `, onlyOn64 : true` missing on `Mac`

File: src/cloudkit.cs [hunk 10]
```diff
@@ -1055,21 +1121,62 @@ interface CKOperation {
...
+		CKOperationGroup Group { get; set; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64 : true), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor] 
+	interface CKOperationGroup : NSSecureCoding {
+
+		[Export ("operationGroupID")]
+		string OperationGroupID { get; }
```
Comment (spouliot) [inline src/cloudkit.cs]: `Id` -> it's not an acronym

File: src/cloudkit.cs [hunk 11]
```diff
@@ -1503,6 +1610,38 @@ interface CKNotificationInfo : NSSecureCoding, NSCopying, NSCoding {
...
+		[NullAllowed, Export ("subtitleLocalizationArgs", ArgumentSemantic.Copy)]
+		string[] SubtitleLocalizationArgs { get; set; }
+
+		[Mac (10, 13), iOS (11, 0)]
+		[Export ("shouldSendMutableContent")]
+		bool ShouldSendMutableContent { get; set; }
+
+		[Mac (10, 13), iOS (11, 0)]
+		[NullAllowed, Export ("collapseIDKey")]
+		string CollapseIDKey { get; set; }
```
Comment (spouliot) [inline src/cloudkit.cs]: `Id`

File: src/CloudKit/Enums.cs [hunk 12]
```diff
@@ -215,4 +216,18 @@ public enum CKQuerySubscriptionOptions : nuint
...
+	public enum CKOperationGroupTransferSize : nint
+	{
+		Unknown,
+		Kilobytes,
+		Megabytes,
+		TensOfMegabytes,
+		HundredsOfMegabytes,
+		Gigabytes,
+		TensOfGigabytes,
+		HundredsOfGigabytes
```
Comment (rolfbjarne) [inline src/CloudKit/Enums.cs]: Minor: missing comma at the end of the last element

File: src/cloudkit.cs [hunk 13]
```diff
@@ -55,6 +55,7 @@ interface CKUserIdentityLookupInfo : NSSecureCoding, NSCopying
 
 		[NullAllowed, Export ("userRecordID", ArgumentSemantic.Copy)]
 		CKRecordID UserRecordID { get; }
+
```
Comment (rolfbjarne) [inline src/cloudkit.cs]: Minor: unnecessary whitespace change.

File: src/cloudkit.cs [hunk 14]
```diff
@@ -1055,21 +1121,62 @@ interface CKOperation {
...
+				   Deprecated = Platform.iOS_11_0 | Platform.Mac_10_13 | Platform.Watch_4_0 | Platform.TV_11_0, 
+			Message = "Instead use CKOperationConfiguration.")]
 		double TimeoutIntervalForResource { get; set; }
 
 		[iOS (9,3)][Mac (10,11,4)]
 		[TV (9,2)]
 		[NullAllowed]
 		[Export ("longLivedOperationWasPersistedBlock", ArgumentSemantic.Strong)]
-		Action LongLivedOperationWasPersistedCallback { get; set; }		
+		Action LongLivedOperationWasPersistedCallback { get; set; }
```
Comment (rolfbjarne) [inline src/cloudkit.cs]: Minor: unnecessary whitespace change.

File: src/cloudkit.cs [hunk 15]
```diff
@@ -1656,7 +1795,12 @@ interface CKFetchDatabaseChangesOperation
 		[NullAllowed, Export ("changeTokenUpdatedBlock", ArgumentSemantic.Copy)]
 		Action<CKServerChangeToken> ChangeTokenUpdated { get; set; }
 
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[NullAllowed, Export ("recordZoneWithIDWasPurgedBlock", ArgumentSemantic.Copy)]
+		Action<CKRecordZoneID> WasPurged { get; set; }
+
 		[NullAllowed, Export ("fetchDatabaseChangesCompletionBlock", ArgumentSemantic.Copy)]
 		CKFetchDatabaseChangesCompletionHandler ChangesCompleted { get; set; }
+
```
Comment (rolfbjarne) [inline src/cloudkit.cs]: Minor: whitespace only change.

File: src/cloudkit.cs [hunk 16]
```diff
@@ -1503,6 +1610,38 @@ interface CKNotificationInfo : NSSecureCoding, NSCopying, NSCoding {
 		[iOS (9,0)][Mac (10,11)]
 		[NullAllowed, Export ("category")]
 		string Category { get; set; }
+
+		[NoTV, Mac (10, 13), iOS (11, 0)]
```
Comment (dalexsoto) [inline src/cloudkit.cs]: `, onlyOn64 : true` missing on `Mac`

Comment (dalexsoto) [reply src/cloudkit.cs]: and also the ones below.

**Comment by spouliot** [comment]
@mandel-macaque ```/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/src/cloudkit.cs(983,64): error CS1525: Unexpected symbol `]', expecting `)' or `,'```

**Comment by mandel-macaque** [comment]
I'm stupid, sorry for that.
